packages/be/crons/cid-importer.js

```diff
@@ -0,0 +1,231 @@
+/**
+ *
+ * ⏱️️ [Cron | every 10 seconds] CID Importer
+ *
+ * Caches CID data
+ *
+ */
+
+// ///////////////////////////////////////////////////// Imports + general setup
+// -----------------------------------------------------------------------------
+const ModuleAlias = require('module-alias')
+const Path = require('path')
+const Axios = require('axios')
+const Fs = require('fs-extra')
+const Express = require('express')
+const Util = require('util')
+const Stream = require('stream')
+const Pipeline = Util.promisify(Stream.pipeline)
+const Spawn = require('child_process').spawn
+require('dotenv').config({ path: Path.resolve(__dirname, '../.env') })
+
+const MC = require('../config')
+
+// ////////////////////////////////////////////////////////////////// Initialize
+MC.app = Express()
+
+// ///////////////////////////////////////////////////////////////////// Aliases
+ModuleAlias.addAliases({
+  '@Root': MC.packageRoot,
+  '@Modules': `${MC.packageRoot}/modules`
+})
+
+try {
+  const modulesRoot = `${MC.packageRoot}/modules`
+  const items = Fs.readdirSync(modulesRoot)
+  items.forEach((name) => {
+    const path = `${modulesRoot}/${name}`
+    if (Fs.statSync(path).isDirectory()) {
+      const moduleName = `${name[0].toUpperCase() + name.substring(1)}`
+      ModuleAlias.addAlias(`@Module_${moduleName}`, path)
+    }
+  })
+} catch (e) {
+  console.log(e)
+}
+
+// ///////////////////////////////////////////////////////////////////// Modules
+require('@Module_Database')
+require('@Module_Cid')
+const { GetFileFromDisk } = require('@Module_Utilities')
+
+// /////////////////////////////////////////////////////////////////// Functions
+// ------------------------------------------------------------ retrieveCidFiles
+const retrieveCidFiles = async (uploads) => {
+  try {
+    const len = uploads.length
+    const cidFileRefs = []
+    for (let i = 0; i < len; i++) {
+      const upload = uploads[i]
+      // if a file already exists with this name in the temp folder, 
+      // delete it to make way for an updated version
+      await deleteTemporaryFile(upload.name) 
+      // fetch file using upload cid
+      const response = await Axios.get(`https://${upload.cid}.ipfs.w3s.link/`, {
+        responseType: 'stream'
+      })
+      await Pipeline(response.data, Fs.createWriteStream(`${MC.packageRoot}/tmp/cid-files/${upload.name}`))
```

Comment:
@svvimming what happens if a file with this exact file name was already written? It doesn't look like we're checking the upload date (set by the `created` and `updated` top-level keys) and selecting the newest one?

Reply:
@orvn so there's a check a couple lines before (on line 62) to see if the file exists and if it does, delete it. maybe that's not the flow we want?

Reply:
I think we should talk about this on a call 🙂